pkg/apiclient/watch-intermediary.go

```diff
@@ -11,20 +11,20 @@ type watchIntermediary struct {
 	events chan *workflowpkg.WorkflowWatchEvent
 }
 
+func (w watchIntermediary) Send(e *workflowpkg.WorkflowWatchEvent) error {
+	w.events <- e
+	return nil
+}
+
 func (w watchIntermediary) Recv() (*workflowpkg.WorkflowWatchEvent, error) {
 	select {
 	case e := <-w.error:
 		return nil, e
-	default:
-		return <-w.events, nil
+	case event := <-w.events:
```

Comment:
Perhaps take a look at the the `open` bool? Perhaps the watch is getting closed
```suggestion
	case event, open := <-w.events:
```